jOOQ/src/main/java/org/jooq/impl/DefaultRecordMapper.java

```diff
@@ -402,7 +402,9 @@ public int compare(Constructor<E> c1, Constructor<E> c2) {
 
                     for (int i = 0; i < parameterTypes.length; i++) {
                         Reflect parameter = Reflect.on(parameters.get(i));
-                        parameterNames.add(parameter.call("getName").<String>get());
+                        // [#8004] Clean up kotlin field name for boolean types
+                        String name = parameter.call("getName").<String>get();
+                        parameterNames.add(getPropertyName(name));
```

Comment:
Thank you very much for your suggestion. `Tools.getPropertyName()` extracts the property name `xyz` from getters of the form `isXyz()` and `getXyz()` as well as from setters of the form `setXyz()`. This seems to have the unwanted side effect of translating too many potential property names. Also, surely we should add a check to make sure this is about a boolean type. I assume that a property of the form `isXyz: String` does not follow the same rules in Kotlin?

Reply:
thanks for the feedback, i'll go ahead and make some updates to my pr